runtime/devnet/src/extensions/mod.rs

```diff
@@ -169,19 +166,24 @@ where
 {
 	const LOG_PREFIX: &str = " read_state |";
 
-	// Prefix params with version, pallet, index to simplify decoding.
+	// Prefix params with version, pallet, index to simplify decoding, and decode parameters for
+	// reading state.
 	params.insert(0, version);
 	params.insert(1, pallet_index);
 	params.insert(2, call_index);
 	let key = <VersionedStateRead<T>>::decode(&mut &params[..])
 		.map_err(|_| DispatchError::Other("DecodingFailed"))?;
 
+	// Charge weight for doing one storage read.
+	env.charge_weight(T::DbWeight::get().reads(1_u64))?;
 	let result = match key {
-		VersionedStateRead::V0(key) => match key {
-			RuntimeRead::Fungibles(key) => read_fungibles_state::<T, E>(key, env),
+		VersionedStateRead::V0(key) => {
+			ensure!(AllowedApiCalls::contains(&key), DispatchError::Other("UnknownCall"));
```

Comment:
Might be better to define a function on DispatchError::unknown_call() rather than these static definitions everywhere.

Reply:
Great suggestion

Reply:
Used constants in stead of functions.